tests/ui/layout/unconstrained-param-ice-137308.rs

```diff
@@ -3,14 +3,16 @@
 //! This used to ICE in layout computation, because `<u8 as A>::B` fails to normalize
 //! due to the unconstrained param on the impl.
 
-#![feature(rustc_attrs)]
+#![feature(rustc_attrs, sized_hierarchy)]
 #![crate_type = "lib"]
 
+use std::marker::PointeeSized;
+
 trait A {
     const B: usize;
 }
 
-impl<C: ?Sized> A for u8 { //~ ERROR: the type parameter `C` is not constrained
+impl<C: PointeeSized> A for u8 { //~ ERROR: the type parameter `C` is not constrained
```

Comment:
did you search and replace the test suite for `?Sized`? I'm not sure what's best, but I would have assumed keeping most of the test suite on `?Sized` and only changing what needs changing would be best.

Reply:
I didn't. In this test, for example, `?Sized` would no longer be unconstrained, it would be `MetaSized`-constrained, and so the test would be doing something different (and failing as a consequence) - I only changed tests that failed and needed changing.

Reply:
I think something is wrong then. `C` is unconstrained because it is not used in the impl, what bounds are on it are irrelevant.

Reply:
I can check if this test still fails because it might just have been the state of the implementation at the time.

Reply:
I've revisited these tests and they fail with some additional "type annotations needed" errors, which I expect is what I was trying to avoid, but these are likely just downstream of the unconstrained parameter error as there's now an unconstrained `C: MetaSized` instead of bare `C`. I've just blessed these instead in the commit which was changing them.